src/OrleansServiceBus/Providers/Streams/EventHub/EventHubCacheEvictionStrategy.cs

```diff
@@ -125,18 +129,18 @@ private bool ShouldPurge(ref CachedEventHubMessage cachedMessage, ref CachedEven
         private void OnFreeBlockRequest(IDisposable block)
         {
             var purgeCandidate = block as FixedSizeBuffer;
-            //free all blocks before purgeCandidate and purgeCandidate
+            //free all blocks before purgeCandidate,including purgeCandidate, expcept for current buffer in use
             if (this.purgedBuffers.Contains(purgeCandidate))
             {
                 while (true)
                 {
-                    var purgedBuffer = this.purgedBuffers.Dequeue();
-                    if (purgedBuffer == purgeCandidate)
-                    {
-                        purgedBuffer.Dispose();
+                    var purgedBuffer = this.purgedBuffers.Peek();
+                    if (purgedBuffer == this.currentBuffer)
```

Comment:
@jason-bragg  went with your suggestion, since my approach will introduce bug in some extreme rare cases